core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala

```diff
@@ -87,4 +145,33 @@ class HttpLineageDispatcher(splineServerRESTEndpointBaseURL: String, http: BaseH
       case _ => Unit
     }
   }
+
+  private def isServerCompressionReady(): Boolean = {
```

Comment:
Why not combining this with the `ensureProducerReady` method? It seems like a duplicated HTTP call.

Reply:
To be backwards compatible we need to support HEAD because GET may fail. We could use HEAD only as a fallback though.

Reply:
But we still can do it in one place, right? We try GET first and recover with HEAD. I don't see any reason why should we postpone the call till the time we are about sending captured lineage data out.